### ISSUE TITLE
JSDoc are applied to wrong schema

### ISSUE THREAD

**Issue by bisubus** [fabien0102/ts-to-zod#54]
## Bug description

Checks that correspod to JSDoc are always applied to outer schema. They likely need to be applied to the most outer schema where they make sense, or several inner schemas if this makes sense. There's currently no difference between `minimum` and `minLength` but it's possible there should be.

### Input

```ts
export interface A {
  /** @minimum 0 */
  a: number | null;
  /** @minLength 1 */
  b: string | null;
  /** @pattern ^c$ */
  c: string | null;
  /** @format email */
  d: string[];
  /** @min 1 */
  e: number[];
  /** @minLength 1 */
  f: string[];
}
```

### Expected output

```ts
export const aSchema = z.object({
  a: z.union([z.number().min(0), z.null()]),
  b: z.union([z.string().min(1), z.null()]),
  c: z.union([z.string().regex(/^c$/), z.null()]),
  d: z.array(z.string().email()),
  e: z.array(z.number()).min(1),
  f: z.array(z.string()).min(1),
});
```

### Actual output

```ts
export const aSchema = z.object({
  a: z.union([z.number(), z.null()]).min(0), // wrong
  b: z.union([z.string(), z.null()]).min(1), // wrong
  c: z.union([z.string(), z.null()]).regex(/^c$/), // wrong
  d: z.array(z.string()).email(), // wrong
  e: z.array(z.number()).min(1), // wrong
  f: z.array(z.string()).min(1), // ok
});
```

**Versions**

- Typescript: `v4.2.3`
- Zod: `v1.6.0`

**Comment by fabien0102** [comment]
Good catch, indeed some cases looks very broken 🙈  

PS: Sorry for the late answer, this last month was a bit busy for me.
PS2: carreful, your version of `zod` looks out-of-date (the generator assume the `v3.9.8` as minimal, so you could have some surprises

**Comment by fabien0102** [comment]
I did fix all the nullable cases (#57), but for the arrays, thinking a bit more about this, adding some jsdoc like those examples:
```ts
/** @minimum 1 */
e: number[];
/** @minLength 1 */
f: string[];
```
should always apply to the top level (and in this case, this will produce some broken result, since `@min 1` can’t really be applied to an array.

For solving this, and removing any possible guess/confusions, I’m proposing this:
```ts
/**
 * @minimum 0
 */
type PositiveNumber = number;

interface A {
  myArray: PositiveNumber[];
}
```

Of course, I just tried and spot an other issue, the `@minimum` is not applied at the top level `type`…

Anyway, thanks again for the reporting! 👍

**Comment by fabien0102** [comment]
@bisubus This should be fixed in `v1.7.0`, all the nullable are correctly translated and you can define subtype with annotations for defining arrays with validations.

Thanks again for raising this issue 😀
